inference-engine/src/vpu/graph_transformer/src/stages/rnn.cpp

```diff
@@ -158,35 +174,73 @@ void FrontEnd::parseRNN(const Model& model, const ie::CNNLayerPtr& _layer, const
 
         RNNRelayout(
             origWeights,
+
             newWeightsPtr,
             newWeightsPtr + ngates * stateSize * inputSize,
+
             ngates,
             stateSize,
             inputSize);
     };
 
     auto newWeights = model->addConstData(_layer->name + "@weights", weights->desc(), generator);
 
-    auto stateCellFinal = model->addFakeData();
-    auto stage = model->addNewStage<LSTMCellStage>(
+    if (outputs.size() == 1) {
+        auto stateCellFinal = model->addFakeData();
+        auto stage = model->addNewStage<LSTMCellStage>(
         layer->name,
         StageType::LSTMCell,
         layer,
         {inputs[0], inputs[1], inputs[2], newWeights, biases},
         {outputs[0], stateCellFinal});
 
-    if (nCells > 1)
-        model->addTempBuffer(stage, DataDesc({stateSize}));
+        if (nCells > 1)
+            model->addTempBuffer(stage, DataDesc({stateSize}));
 
-    bool RNNForward = layer->direction == ie::RNNSequenceLayer::FWD;
-    stage->attrs().set<bool>("RNNForward", RNNForward);
-    stage->attrs().set<int>("nCells", nCells);
-    stage->attrs().set<int>("nBatches", nBatches);
+        bool RNNForward = layer->direction == ie::RNNSequenceLayer::FWD;
+        stage->attrs().set<bool>("RNNForward", RNNForward);
+        stage->attrs().set<int>("nCells", nCells);
+        stage->attrs().set<int>("nBatches", nBatches);
+    }
+
+    if (outputs.size() == 2) {
+        auto stage = model->addNewStage<LSTMCellStage>(
+        layer->name,
+        StageType::LSTMCell,
+        layer,
+        {inputs[0], inputs[1], inputs[2], newWeights, biases},
+        {outputs[0], outputs[1]});
+
+        if (nCells > 1)
+            model->addTempBuffer(stage, DataDesc({stateSize}));
+
+        bool RNNForward = layer->direction == ie::RNNSequenceLayer::FWD;
+        stage->attrs().set<bool>("RNNForward", RNNForward);
+        stage->attrs().set<int>("nCells", nCells);
+        stage->attrs().set<int>("nBatches", nBatches);
+    }
+
+    if (outputs.size() == 3) {
```

Comment:
the difference between ouputs.size() == 2 and outputs.size()==3 is only outputs? So why do you duplicate the other code? you can create ouputs and fill it in accordance with outputs number.